antismash/modules/rrefinder/test/test_rrefinder.py

```diff
@@ -0,0 +1,297 @@
+        assert res_object.bitscore_cutoff == self.bitscore_cutoff
+        assert res_object.min_length == self.min_length
+        assert res_object.record_id == self.record_id
+
+        json_dict2 = dict(self.json_dict)
+        json_dict2['record_id'] = 'another_record'
+        assert not RREFinderResults.from_json(json_dict2, self.record, self.min_length, self.bitscore_cutoff)
+
+        json_dict3 = dict(self.json_dict)
+        json_dict3['schema_version'] = 'not_a_valid_schema'
+        assert not RREFinderResults.from_json(json_dict3, self.record, self.min_length, self.bitscore_cutoff)
+
+        # Invalid arguments should return None
+        json_dict4 = dict(self.json_dict)
+        json_dict4.pop('bitscore_cutoff')
+        with self.assertRaises(ValueError):
+            RREFinderResults.from_json(json_dict4, self.record, self.min_length, self.bitscore_cutoff)
+
+        json_dict5 = dict(self.json_dict)
+        json_dict5.pop('min_length')
+        with self.assertRaises(ValueError):
+            RREFinderResults.from_json(json_dict5, self.record, self.min_length, self.bitscore_cutoff)
+
+        # More lenient settings shouldn't return results
+        assert not RREFinderResults.from_json(self.json_dict, self.record, 25, self.bitscore_cutoff)
+        assert not RREFinderResults.from_json(self.json_dict, self.record, self.min_length, 15.0)
+
+        # Stricter settings should filter
+        res_object2 = RREFinderResults.from_json(self.json_dict, self.record, self.min_length, 35.0)
+        assert res_object2.hit_info['locus_tag_a'] == [self.hit_a1]
+        assert len(res_object2.hit_info) == 1
+        assert res_object2.hits_per_protocluster[1] == ['locus_tag_a']
+        assert len(res_object2.hits_per_protocluster) == 1
+
+        res_object3 = RREFinderResults.from_json(self.json_dict, self.record, 80, self.bitscore_cutoff)
+        assert res_object3.hits_per_protocluster[2] == ['locus_tag_b']
+        assert len(res_object3.hits_per_protocluster) == 1
+        assert res_object3.hit_info['locus_tag_b'] == [self.hit_b1]
+        assert len(res_object3.hit_info) == 1
+
+    def test_add_record(self):
+        with self.assertRaisesRegex(ValueError, "Record to store in and record analysed don't match"):
+            self.res_object.add_to_record(self.record2)
+        self.res_object.add_to_record(self.record)
+        assert self.record.add_feature.call_count == 2
+        for arg in self.record.add_feature.call_args_list:
+            # First argument of positional arguments
+            assert isinstance(arg[0][0], RRE)
+
+
+class TestRREFinder(unittest.TestCase):
+    def setUp(self):
+        self.ripps = ['bacteriocin', 'cyanobactin', 'lanthipeptide',
+             'lassopeptide', 'linaridin', 'thiopeptide', 'sactipeptide',
+              'proteusin', 'glycocin', 'bottromycin', 'microcin']
+
+        self.hit_a1 = {'locus_tag': 'locus_tag_a', 'location': '[1000:2000]',
```

Comment:
Seeing all these hit dictionaries makes me realise they should be a class, but I'm happy to do that after (or before and I'll convert your usage of the dictionary version into the class one).

Reply:
That makes sense to me. The hits dictionaries made here are have the same organisation as used in the HmmerResults object, so I guess this would require a larger rework?